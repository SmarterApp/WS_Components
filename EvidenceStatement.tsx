import * as React from 'react';

interface Props {
    statement: string;
}

export class EvidenceStatement extends React.Component<Props, {}> {
    render() {
        return (
            <div className="centered-table-container">
                <table className='item-data-table'>
                    <thead>
                        <tr>
                            <th>Evidence Statement</th>
                        </tr>
                    </thead>
                    <tbody>
                        <tr>
                            <td>{this.props.statement}</td>
                        </tr>
                    </tbody>
                </table>
            </div>
        );
    }
}