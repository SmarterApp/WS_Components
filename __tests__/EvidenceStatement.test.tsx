import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { EvidenceStatement } from '../EvidenceStatement';
import { shallow } from 'enzyme';

describe("EvidenceStatement", () => {
    it("matches snapshot", () => {
        const wrapper = shallow(<EvidenceStatement statement="" />)
        expect(wrapper).toMatchSnapshot();
    })
})
