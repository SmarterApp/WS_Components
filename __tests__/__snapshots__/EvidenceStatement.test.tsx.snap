// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvidenceStatement matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className="centered-table-container"
>
    <table
        className="item-data-table"
    >
        <thead>
            <tr>
                <th>
                    Evidence Statement
                </th>
            </tr>
        </thead>
        <tbody>
            <tr>
                <td />
            </tr>
        </tbody>
    </table>
</div>,
  "nodes": Array [
    <div
      className="centered-table-container"
>
      <table
            className="item-data-table"
      >
            <thead>
                  <tr>
                        <th>
                              Evidence Statement
                        </th>
                  </tr>
            </thead>
            <tbody>
                  <tr>
                        <td />
                  </tr>
            </tbody>
      </table>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <EvidenceStatement
        statement=""
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": EvidenceStatement {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "statement": "",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className="centered-table-container"
>
          <table
                    className="item-data-table"
          >
                    <thead>
                              <tr>
                                        <th>
                                                  Evidence Statement
                                        </th>
                              </tr>
                    </thead>
                    <tbody>
                              <tr>
                                        <td />
                              </tr>
                    </tbody>
          </table>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className="centered-table-container"
>
          <table
                    className="item-data-table"
          >
                    <thead>
                              <tr>
                                        <th>
                                                  Evidence Statement
                                        </th>
                              </tr>
                    </thead>
                    <tbody>
                              <tr>
                                        <td />
                              </tr>
                    </tbody>
          </table>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <EvidenceStatement
    statement=""
/>,
}
`;
